Tongues

Incoming5643 opened the pull request:
Adds tongues, which serve as the physical manifestation of speech impediments.

Sprites by @Cheridan 

_"A Super Aggro Crag Request Pull"_™

:cl: Incoming5643
rscadd: Leading scientists have finally proven that humans (and xenos) in fact have tongues in their mouths.
rscadd: Swapping out tongues allows the various races to change their speech impediments. Putting a xeno tongue in someone allows them to make the hissing sound like a xeno whenever they speak.
experiment: Remember that tongues are stored in the MOUTH, selecting the head isn't good enough.
/:cl:

Incoming5643 committed:
Tongues WIP

Adds tongues, which serve as the physical manifestation of speech impediments.

RemieRichards commented:
abductors don't have mouths nor tongues nerd.

Incoming5643 commented:
Read the damn commit I've not broken keyfabe

RemieRichards commented on code/modules/mob/living/carbon/human/species.dm:
```diff
@@ -127,6 +127,10 @@
 		C.reagents.add_reagent(exotic_blood, 80)
+	for(var/path in mutant_organs)
+		var/obj/item/organ/internal/I = new path()
+		I.owner = C
```
why is this not ```I.Insert(C)```

Incoming5643 replied:
No idea, might explain some things.

GunHog commented:
Add a changeling one that you can give to people to grant them the ability to suck people dry.

Incoming5643 commented:
Tying ling abilities to organ systems the crew can steal is a long term goal of mine, but organ system code isn't quite there yet.

Incoming5643 committed:
Remie catching me on real coder hours

Core0verload commented:
Why zone head when you can actually put them in mouth? It's supported by some of other current organs.

Core0verload commented on code/modules/mob/living/carbon/human/species.dm:
```diff
@@ -41,7 +41,7 @@
 	var/list/default_features = list() // Default mutant bodyparts for this species. Don't forget to set one for every mutant bodypart you allow this species to have.
 	var/list/mutant_bodyparts = list() 	// Parts of the body that are diferent enough from the standard human model that they cause clipping with some equipment
-
+	var/list/mutant_organs = list(/obj/item/organ/internal/tongue)		//Internal organs that are unique to this race.
```
Why `zone = "head"` when you can actually put them in mouth? It's supported by some of existing organs and organ manipulation for mouth is already a thing.

Core0verload commented on code/modules/surgery/organs/organ_internal.dm:
```diff
@@ -239,6 +239,95 @@
+	name = "tongue"
+	desc = "A fleshy muscle mostly used for lying."
+	icon_state = "lungs"
+	zone = "head"
```
Why `zone = "head"` when you can actually put them in mouth? It's supported by some of existing organs and organ manipulation for mouth is already a thing.

Core0verload commented:
At least one of the ling abilities is already tied to organ - it's thermal vision I think.

Fox-McCloud commented:
@Core0verload 

More abilities should be, IMO---like hivemind, stasis, etc--right now most are just tied to the status of being a changeling or not.

coiax commented:
So this means there's a tongue organ that you can remove? People can still talk without tongues, just deprives them of a lot of sounds

octareenroon91 commented:
So loss of tongue gives an entire new impediment?

Alek2ander commented:
Suiciding by biting off your tongue should drop the tongue.
Also, this is something that definitely needs food recipes.

AnturK commented:
:+1: for the idea. Keep the abductor team separation please though.

ChangelingRain commented on code/modules/surgery/organs/organ_internal.dm:
```diff
@@ -239,6 +239,95 @@
+	var/rendered = "<i><font color=#800080><b>[user.name]:</b> [message]</font></i>"
+	for(var/mob/living/carbon/human/H in mob_list)
+		var/obj/item/organ/internal/tongue/T = H.getorganslot("tongue")
+		if(!T || T.type != type)
```
You need to check the 'team' of the tongue somehow, so that multiple ships of abductors can't hear eachother. Though I suppose that's kind of a feature.

Incoming5643 commented:
there's no impediment for not having a tongue, you just can't talk.

if someone could figure out how to make a heavily edited accent to represent not having a tongue I'd be down with that though.

coiax commented:
https://www.reddit.com/comments/ifflf/i_have_no_tongue_ama/

> > Can you speak?

> Yes, I can speak fairly well. I had speech therapy for a year and my speech improved a lot, but I still have a lot of trouble with certain sounds. The "G" as in goat, "K" sound and the "ee" sound are the worst. Interestingly, I find that the more intelligent the person, the easier they can understand me.
> For the D, N, T and TH sounds I use a special technique that has really helped me. I press my upper front teeth against my lower lip to produce the start of the sound. Depending on the sound, I vary the pressure applied to my lip. T requires the most, then D and least is N.
> For the K sound, I use my throat to pronounce it, much like the Germans produce very throaty sounds in their laguage.
> Finally, the G in goat is impossible for me to say and I have never been able to produce it.

Supermichael777 commented on code/modules/surgery/organs/organ_internal.dm:
```diff
@@ -239,6 +239,95 @@
+
+/obj/item/organ/internal/tongue/abductor
+	name = "superlingual matrix"
+	desc = "A mysterious structure that allows for instantaneous and omnipresence communication between users. Pretty impressive until you need to eat something."
```
omnipresent

coiax replied:
also, I think I've seen brain damaged abductors actually gibber out loud.

Zilenan91 commented:
As for organ plans big :-1:  to letting crewmembers hear any kind of hivemind, this includes xeno and changeling, but everything else is fine.

coiax commented:
But crew members can already hear the xeno hivemind with organ transplants, and binary chat with the binary encryption key.

Zilenan91 commented:
The binary key is fine because the borgs can tell everyone over comms if they want to and aren't immediately hostile.

Also I already said the xeno hivemind transplant, and it's the stupidest shit ever because Xenos are complete shit and letting people hear their plans and wherever they're going when one of them dies is just depressingly stupid game design combined with them bizarrely being less durable than even regular humans.

Incoming5643 committed:
Generalize the system down to carbon, adding a xeno tongue that handles the actual hissing sound.
Moves the tongue organ to the mouth
Fixes up synth inheritances
Losing your tongue no longer makes you silent, but rather just makes you a lot harder to understand
Little misc things

Incoming5643 commented:
Could be merged today save for lack of sprites

Ultimate-Chimera commented:
Xeno tongue-mouth please. ;)

PKPenguin321 commented:
xeno ;) tounge ;)i;)n ;;;)))my ;;)) ass ;;;;))))please. ;;;;;;;;;))))))))))

Cheridan commented:
http://i.imgur.com/cicjfPY.png

https://www.dropbox.com/s/fol64t8eyupy49z/tongues.dmi?dl=0

Incoming5643 committed:
Implements Cheridan's tongue sprites. Oh baby.

Incoming5643 commented:
We should be good to go.

ChangelingRain commented:
Mime speaks? RIP OUT THE TONGUE!

Ultimate-Chimera commented:
@ChangelingRain Why did you remove the xeno hissing sound when it talks?

Incoming5643 commented:
eh? They should still hiss, are they not hissing?

Ultimate-Chimera commented:
No, you completely deleted the code for it.

Incoming5643 commented:
it was moved to the xeno tongue

Ultimate-Chimera commented:
Yes, but it doesn't work with normal xenos now. Can you add it back?

Incoming5643 commented:
Xenos have the tongue so it should be working, I'll look into it

Ultimate-Chimera commented:
You removed the hissing sounds from their saycode. They no longer make hissing sounds.

Incoming5643 commented:
But I put it in their tongue, which is called from carbon/treat_message

I'm going to guess that aliens don't call treat_message.

Ultimate-Chimera commented:
I'm making a hotfix. I'm guessing you removed that code because if their tongue is removed they can't hiss anymore?

Incoming5643 commented:
The alien should call treat_message, which should call their tongue, which should produce the hiss.

There's nothing special about how aliens talk compared to other carbons (alien_talk is for hivemind), so maybe it's just a matter of the sound not being heard for some reason.

Ultimate-Chimera commented:
No, it is completely removed. I had to use the *hiss emote to even make a hissing sound.

Ultimate-Chimera commented:
#16864

Niknakflak commented on code/modules/surgery/organs/organ_internal.dm:
```diff
@@ -239,6 +239,110 @@
+/obj/item/organ/internal/tongue/abductor
+	name = "superlingual matrix"
+	desc = "A mysterious structure that allows for instant communication between users. Pretty impressive until you need to eat something."
+	icon_state = "tongueayylmao"
```
pff

Ultimate-Chimera replied:
What's this for?
